code/modules/games/gamehelm.dm

```diff
@@ -8,103 +8,131 @@
 	desc_info = "You can ALT-click the game-helm to open it up and turn it on. Click on the open device to play."
 	icon = 'icons/obj/gamehelm.dmi'
 	w_class = ITEMSIZE_SMALL
-	var/open = "open_white"
-	var/closed = "closed_white"
+	update_icon_on_init = TRUE
 
-/obj/item/gamehelm/Initialize()
-	. = ..()
-	icon_state = closed
-	add_overlay("buttons_closed")
+	var/case_color = "white"
+	var/open = FALSE
 
-/obj/item/gamehelm/AltClick()
-	if(use_check(usr))
-		return
-	if(icon_state == open) //Toggle the lid, turn it off.
-		icon_state = closed
-		cut_overlays()
-		add_overlay("buttons_closed")
-	else if(icon_state == closed) //Otherwise open it up
-		icon_state = open
-		cut_overlays()
+	var/playing_game = FALSE
+	var/current_screen_state = null
+	var/static/list/game_type_to_state = list(
+		"anime game" = "screen_anime",
+		"shooter game" = "screen_shooter",
+		"strategy game" = "screen_strategy",
+		"piloting game" = "screen_pilot",
+		"horror game" = "screen_dread",
+		"simulation game" = "screen_emotive",
+		"exploration game" = "screen_exploration",
+		"music service" = "screen_music",
+		"video service" = "screen_video",
+		"turn off the system" = null
+	)
+	var/static/list/non_game_states = list(
+		"screen_music",
+		"screen_video"
+	)
+
+	var/static/list/game_actions = list(
+		"hits all the buttons at the same time" = "hit all the buttons at the same time",
+		"rapidly hits a bunch of buttons" = "rapidly hit a bunch of buttons",
+		"tilts the InUs Game-Helm to the right" = "tilt the InUs Game-Helm to the right",
+		"tilts the InUs Game-Helm to the left" = "tilt the InUs Game-Helm to the left",
+		"swipes across the InUs Game-Helm's screen" = "swipe across the InUs Game-Helm's screen"
+	)
+
+/obj/item/gamehelm/update_icon()
+	cut_overlays()
+	if(open)
+		icon_state = "open_[case_color]"
 		add_overlay("buttons_open")
+	else
+		icon_state = "closed_[case_color]"
+		add_overlay("buttons_closed")
+	if(current_screen_state)
+		add_overlay(current_screen_state)
+
+/obj/item/gamehelm/process()
+	if(playing_game)
+		playsound(loc, /singleton/sound_category/quick_arcade, 45)
+		if(ismob(loc))
+			var/picked_action = pick(game_actions)
+			var/self_action = game_actions[picked_action]
+			loc.visible_message("<b>[loc]</b> [picked_action]!", SPAN_NOTICE("You [self_action]!"), range = 3)
```

Comment:
this will get spammy very quick for the player and anyone nearby. make it a toggle and/or do something with world.time so it happens once in a while.